src/tlo/methods/hsi_event.py

```diff
@@ -187,9 +190,54 @@ def _run_after_hsi_event(self) -> None:
 
     def run(self, squeeze_factor):
         """Make the event happen."""
+        
+        print_chains = False
+        df_before = []
+        
+        if self.sim.generate_event_chains:
```

Comment:
obviously would be nice to factorise-out this logic, which repeats in events.py

(Shame we don't have HSI_Event inheriting from Event, and we'd get it for free. We used to.. but it was changed at some point for a reason I can no longer remember.)